src/Plutarch/Extra/Ord.hs

```diff
@@ -175,6 +200,243 @@ instance DerivePlutusType (PComparator a) where
 
 -- TODO: Semigroup, Monoid
 
+{- | Similar to 'PList', but ensures its contents remain sorted according to the
+ 'POrd' instance of @a@.
+
+ = Note
+
+ We do /not/ derive 'Generic' for this type, as that would leak its
+ internals.
+
+ = Note on conversions
+
+ As 'PSortedList' is not able to be an instance of 'PListLike', if you want to
+ convert to 'PList' specifically, use 'pupcast' instead.
+
+ @since 3.6.1
+-}
+newtype PSortedList (a :: S -> Type) (s :: S)
+    = PSortedList (Term s (PList a))
+
+-- | @since 3.6.1
+instance PlutusType (PSortedList a) where
```

Comment:
That requires a `Generic` derivation, which leaks internals everywhere.

Reply:
Oh...

Reply:
It's one of my big gripes against `Generic`. Once you derive it, all internal invariants are now purely suggestions, since someone can construct a `Rep` by hand, then force it to be read as your type, invariants be damned.